pyteal/ast/frame.py

```diff
@@ -86,16 +178,19 @@ def has_return(self) -> bool:
 
 
 class FrameBury(Expr):
-    def __init__(self, value: Expr, frame_index: int):
-        from pyteal.ast.subroutine import SubroutineCall
-
+    def __init__(
+        self,
+        value: Expr,
+        frame_index: int,
+        *,
+        validate_types: bool = True,
```

Comment:
I don't yet see the value of exposing the `validate_types` parameter. Perhaps it is utilized in the next PR?

Reply:
okay, circling back to this discussion.

My thinking is, `FrameBury` should be only used by `FrameVar`, through `store` method.
- `FrameBury` used to be oblivious of the type of the stack position to bury to, now we can use `ProtoStackLayout` to tell the exact `TealType`.
- The additional knowledge of `TealType` allows `FrameBury` to validate type, so I extend the option `validate_types` to `FrameBury`, so it validates the type of argument that it construct from.
- The final type checking does not need to replicate 2 pieces of similar logic as you pointed out before: https://github.com/algorand/pyteal/pull/584#discussion_r1014341184, so I see it as somehow an improvement. The final implementation of `FrameVar.store` should look somehow like `ScratchVar.store`.